pages: setting a GRUB password (#429)
coreos/fedora-coreos-docs: modules/ROOT/pages/grub-password.adoc

= Setting a GRUB password

> Some FCOS users may want to setup a password to prevent unauthorized users from accessing the GRUB command line, modifying kernel command-line arguments, or booting non-default OSTree deployments. The steps below will outline the process of setting a GRUB password.

Reviewer: ```suggestion
You can setup a GRUB password to prevent unauthorized users from accessing the GRUB command line, modifying kernel command-line arguments, or booting non-default OSTree deployments. The process is outlined in the steps below:
```

We may even want to remove the ` The process is outlined in the steps below:` part.

== Creating the password hash

> We will use `grub2-mkpasswd-pbkdf2` to create a password hash for GRUB.

Reviewer: We usually use the "you" form in the docs:
```suggestion
You can use `grub2-mkpasswd-pbkdf2` to create a password hash for GRUB:
```

[source, bash]
----
$ grub2-mkpasswd-pbkdf2
Enter password: <PASSWORD>
Reenter password: <PASSWORD>
PBKDF2 hash of your password is grub.pbkdf2.sha512.10000.5AE6255...
----

NOTE: `grub2-mkpasswd-pbkdf2` tool is a component of the `grub2-tools-minimal` package on Fedora.

== Butane config

> With the password hash ready, we can create our butane config:

Reviewer: Same here

[source, yaml]
----
variant: fcos
version: 1.5.0-experimental
grub:
  users:
    - name: root
      password_hash: grub.pbkdf2.sha512.10000.5AE6255...
----

> We have defined a GRUB superuser `root` and set the password for that user using a hash. At a lower level, this config will generate a `user.cfg` file in `/boot/grub2/` with the necessary GRUB commands to set the given username and password. 

Reviewer: Same here

> We can now use this config to boot a Fedora CoreOS instance.

Reviewer: idem

= Setting a GRUB password

Some FCOS users may want to setup a password to prevent unauthorized users from accessing

> GRUB command line, modifying kernel command-line arguments, or booting non-default OSTree deployments.

Reviewer: ```suggestion
the GRUB command line, modifying kernel command-line arguments, or booting non-default OSTree deployments.
```

The steps below will outline the process of setting a GRUB password.

== Creating the password hash
We will use `grub2-mkpasswd-pbkdf2` to create a password hash for GRUB.

[source, bash]
----
$ grub2-mkpasswd-pbkdf2
Enter password: <PASSWORD>
Reenter password: <PASSWORD>
PBKDF2 hash of your password is grub.pbkdf2.sha512.10000.5AE6255...
----

NOTE: `grub2-mkpasswd-pbkdf2` tool is a component of the `grub2-tools-minimal` package on Fedora.

> == Butane config

Reviewer: Nit: we probably need an empty line after the titles

With the password hash ready, we can create our butane config:

[source, yaml]
----
variant: fcos
version: 1.5.0-experimental
grub:
  users:
    - name: root
      password_hash: grub.pbkdf2.sha512.10000.5AE6255...
----

> We have defined a GRUB superuser `root` and set the password for that user using a hash. On a lower level,

Reviewer: ```suggestion
We have defined a GRUB superuser `root` and set the password for that user using a hash. At a lower level,
```
